backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/repository/OrgChangeRepository.java

```diff
@@ -0,0 +1,734 @@
+				.filter(BPartnerContact::isActive)
+				.filter(contact ->
+						{
+							final BPartnerContactType contactTypeType = contact.getContactType();
+							return contactTypeType.getIsShipToDefaultOr(false);
+						})
+				.findFirst()
+				.orElse(null);
+	}
+
+	@Nullable
+	public BPartnerContact getPurchaseDefaultContactOrNull(final List<BPartnerContact> contacts)
+	{
+		return contacts.stream()
+				.filter(BPartnerContact::isActive)
+				.filter(contact ->
+						{
+							final BPartnerContactType contactTypeType = contact.getContactType();
+							return contactTypeType.getIsPurchaseDefaultOr(false);
+						})
+				.findFirst()
+				.orElse(null);
+	}
+
+	@Nullable
+	public BPartnerContact getSalesDefaultContactOrNull(final List<BPartnerContact> contacts)
+	{
+		return contacts.stream()
+				.filter(BPartnerContact::isActive)
+				.filter(contact ->
+						{
+							final BPartnerContactType contactTypeType = contact.getContactType();
+							return contactTypeType.getIsSalesDefaultOr(false);
+						})
+				.findFirst()
+				.orElse(null);
+	}
+
+	public void unmarkBillToDefaultLocations(final List<BPartnerLocation> locations)
+	{
+		for (final BPartnerLocation location : locations)
+		{
+			final BPartnerLocationType type = location.getLocationType();
+			type.setBillToDefault(false);
+		}
+	}
+
+	public void unmarkShipToDefaultLocations(final List<BPartnerLocation> locations)
+	{
+		for (final BPartnerLocation location : locations)
+		{
+			final BPartnerLocationType type = location.getLocationType();
+			type.setShipToDefault(false);
+		}
+	}
+
+	public void unmarkDefaultContacts(final List<BPartnerContact> contacts)
+	{
+		for (final BPartnerContact contact : contacts)
+		{
+			final BPartnerContactType contactType = contact.getContactType();
+			contactType.setDefaultContact(false);
+		}
+	}
+
+	public void unmarkBillToDefaultContacts(final List<BPartnerContact> contacts)
+	{
+		for (final BPartnerContact contact : contacts)
+		{
+			final BPartnerContactType contactType = contact.getContactType();
+			contactType.setBillToDefault(false);
+		}
+	}
+
+	public void unmarkPurchaseDefaultContacts(final List<BPartnerContact> contacts)
+	{
+		for (final BPartnerContact contact : contacts)
+		{
+			final BPartnerContactType contactType = contact.getContactType();
+			contactType.setPurchaseDefault(false);
+		}
+	}
+
+	public void unmarkSalesDefaultContacts(final List<BPartnerContact> contacts)
+	{
+		for (final BPartnerContact contact : contacts)
+		{
+			final BPartnerContactType contactType = contact.getContactType();
+			contactType.setSalesDefault(false);
+		}
+	}
+
+	public void unmarkShipToDefaultContacts(final List<BPartnerContact> contacts)
+	{
+		for (final BPartnerContact contact : contacts)
+		{
+			final BPartnerContactType contactType = contact.getContactType();
+			contactType.setShipToDefault(false);
+		}
+	}
+
+	public BPartnerLocation createNewLocation(@NonNull final BPartnerLocation existingLocationInInitialPartner)
```

Comment:
wouldn't `existingLocationInInitialPartner.toBuilder().active(true).build()` be better? (same question from here till `createNewBankAccount`) 
* i noticed that this method is called only one time (plus again, i don't think this belongs into a repo), so i'm not sure if we need it at all.
* if new properties are added to `BPartnerLocation` we probably won't think to add them here as well

Reply:
You are right. Changed